build.sbt

```diff
@@ -439,14 +439,7 @@ lazy val tests =
       // nativeOptimizerReporter := OptimizerReporter.toDirectory(
       //   crossTarget.value),
       libraryDependencies += "org.scala-native" %%% "test-interface" % nativeVersion,
-      testFrameworks += new TestFramework("tests.NativeFramework"),
-      envVars in (Test, test) ++= Map(
-        "USER"                           -> "scala-native",
-        "HOME"                           -> baseDirectory.value.getAbsolutePath,
-        "SCALA_NATIVE_ENV_WITH_EQUALS"   -> "1+1=2",
-        "SCALA_NATIVE_ENV_WITHOUT_VALUE" -> "",
-        "SCALA_NATIVE_ENV_WITH_UNICODE"  -> 0x2192.toChar.toString
-      )
```

Comment:
The reason we set those environment variables in the build is that they have to be external to the testing process. If we set those environment variables from within tests, we don't know if getenv/setenv actually interacts with the outer world, or just updates a process-local hash map.

Reply:
The reason I was trying to move this to the test was that it lack encapsulation and it is not DRY this way. I was also trying to move stuff out of the build so we are less dependent on the sbt build. 

Also, the way an env var is set on the JVM is via reflection on a private member to modify the env map. https://stackoverflow.com/questions/318239/how-do-i-set-environment-variables-from-java

Overall, there are quite a few things going on here so it may be better if I break this PR into multiple pieces since I can't demonstrate the problem that easily without some changes.

Typically, the environment in C can be changed for the running process and then is inherited by any child processes. It appears we may be inheriting the env from the Java process as described in the SOF above so it is masking the underlying problem.

Reply:
We used to have a test that checks that environment variables set from outside are visible in native. After this change it's no longer testing this, it's testing that environment variables set from inside native are visible inside native which is not the same thing. This is a regression.

Reply:
I looked into this in `sbt` and it seems that the `envVars` are passed to the spawned process for the tests. I think since there is no way to change the `env` of a running Java process, this was the only way to actually test this at all. If we add a native `setenv`, this updates the current process `env` so we can test this way.

If you are conceptually ok with moving the data to the test area to reduce the dependency on `sbt` and to make the tests DRY then I have started another PR to add the `setenv` and I can open another one to adjust the tests.
See https://github.com/scala-native/scala-native/pull/1151